### DISCUSSION TITLE
GitHub Pages Entry File not working as advertised

### DISCUSSION THREAD

**Discussion by marcus-darden** [community/community#66355] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

The GHP docs say:

"Create the entry file for your site. GitHub Pages will look for an index.html, index.md, or README.md file as the entry file for your site.

If your publishing source is a branch and folder, the entry file must be at the top level of the source folder on the source branch. For example, if your publishing source is the /docs folder on the main branch, your entry file must be located in the /docs folder on a branch called main."

This is what my repo looks like:

```
..
├── README.md
└── docs
    ├── README.md
    └── search.cpp
```

Pages generation is "Deploy from a branch," and set to the 'main' branch, 'docs/' folder, but navigating a browser to link given from the deployment, https://organization.github.io/gitname, results in 404. The only way to get it work is to go to https://organization.github.io/gitname/README.html.

What gives? How do I get the site entry come from `README.md`? My current workaround is to rename `README.md -> index.md`, but that is not what is advertised.

**Comment by queenofcorgis** [comment]
Hey @marcus-darden - do you have two files in your repo named readme.md?

**Reply by marcus-darden** [reply]
Yes, one at the top-level of the repo for contributors, and one in the folder which is the GHP root: docs/

... and neither gets used as the the entry page.

**Reply by queenofcorgis** [reply]
@marcus-darden are you able to share the repo with us - it may help the community better troubleshoot?

**Reply by marcus-darden** [reply]
Public: https://github.com/eecs281staff/search-demo

The currently published version has the docs/README.md file renamed to docs/index.md (the temporary fix I mentioned).

To view the problem, checkout HEAD^ and build for GHP, to notice that the resulting site does not have an entry page.

**Reply by marcus-darden** [reply]
Looks like nothing here...

**Reply by marcus-darden** [reply]
@queenofcorgis Was there anything that came up when researching this inconsistency?
